File: backend/app/schemas/notification.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificationBase(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    title: str
    content: str
    notification_type: str
    entity_type: str | None = None
    entity_id: uuid.UUID | None = None
    is_read: bool = False


class NotificationRead(NotificationBase):
    id: uuid.UUID
    created_at: datetime


class UnreadCountResponse(BaseModel):
    unread_count: int

File: frontend/src/types/notification.ts
export interface Notification {
  id: string;
  title: string;
  content: string;
  notification_type: string;
  entity_type: string | null;
  entity_id: string | null;
  is_read: boolean;
  created_at: string;
}

export interface UnreadCountResponse {
  unread_count: number;
}

export interface MarkAllReadResponse {
  marked_read: number;
}